p2pclient/java/p2pc.java

```diff
@@ -0,0 +1,21 @@
+public class p2pc {
```

Comment:
What is the purpose of having the Java binding in this repo? Is it just sample code, or do we want downstream users to use this as a dependency? If the latter is the case, we should add a Java build system (Maven/gradle) and publish to maven.

Reply:
The thing about the Go binding is that it's usable for downstream users in this form. They just import the `...p2pclient/go` package and that's it. But in the current form, this is not usable for Java users.

Reply:
> What is the purpose of having the Java binding in this repo? Is it just sample code

Good question.  I don't want to pollute the repo with unnecessary code.  I can refactor and make it example code if you all prefer that.

Reply:
> The thing about the Go binding is that it's usable for downstream users in this form. They just import the ...p2pclient/go package and that's it. But in the current form, this is not usable for Java users.

Agreed.  Another option is to move the java specific binding code to a bindings/java folder.  Then I could move the java main to an examples/java folder.

Reply:
@raulk I added all code related to bindings to a bindings folder.  Also, i moved all java specific binding code to bindings/java (the example java classes are in bindings/java/examples).  I think this will help